xilem_core/src/view/memoize.rs

```diff
@@ -81,31 +82,41 @@ where
         &self,
         view_state: &mut Self::ViewState,
         id_path: &[ViewId],
-        message: Box<dyn Any>,
+        message: DynMessage,
         app_state: &mut State,
     ) -> MessageResult<Action> {
-        let r = view_state
-            .view
-            .message(&mut view_state.view_state, id_path, message, app_state);
-        if matches!(r, MessageResult::RequestRebuild) {
+        let message_result =
+            view_state
+                .view
+                .message(&mut view_state.view_state, id_path, message, app_state);
+        if matches!(message_result, MessageResult::RequestRebuild) {
             view_state.dirty = true;
         }
-        r
+        message_result
     }
-}
 
-/// A static view, all of the content of the `view` should be constant, as this function is only run once
-pub fn static_view<V, F>(view: F) -> Memoize<(), impl Fn(&()) -> V>
-where
-    F: Fn() -> V + Send + 'static,
-{
-    Memoize::new((), move |_: &()| view())
+    fn teardown(
+        &self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut Context,
+        element: <Self::Element as crate::ViewElement>::Mut<'_>,
+    ) {
+        view_state
+            .view
+            .teardown(&mut view_state.view_state, ctx, element);
+    }
 }
 
 /// Memoize the view, until the `data` changes (in which case `view` is called again)
-pub fn memoize<D, V, F>(data: D, view: F) -> Memoize<D, F>
+pub fn memoize<State, Action, Context, Data, V, ViewFn>(
+    data: Data,
+    view: ViewFn,
+) -> Memoize<Data, ViewFn>
 where
-    F: Fn(&D) -> V + Send,
+    Data: PartialEq + 'static,
+    ViewFn: Fn(&Data) -> V + 'static,
+    V: View<State, Action, Context>,
+    Context: ViewPathTracker,
```

Comment:
I had this in `xilem` as well with bounds for `WidgetView + Send + Sync` etc. but I decided against it, because I think it's more practical to have just one `memoize` function for different `View` implementations. So that this function can also be used in potential `xilem_web` contexts in `xilem` without having to use a different function.

I've also removed `static_view` mostly because of the return type, and that I think that `Arc` should be probably used instead (only inconvenience is having to use `.clone()` though).